### PR TITLE
MOB-3277 [Config & Prep] Navigation: Implementation of a "lite" burge…

### PR DESCRIPTION
**Pull request by mateid-kr**
…r menu for guest checkout mode

## JIRA
https://karhoo.atlassian.net/browse/MOB-3277

## DESCRIPTION
MOB-3277 [Config & Prep] Navigation : Implementation of a "lite" burger menu for guest checkout mode

## Types of changes

What types of changes does your code introduce to the project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix 
- [X] New feature 
- [ ] Tech Debt
- [ ] Breaking change 
- [ ] Documentation Update
- [ ] Other: *define what type of change it is*

## CHECKLIST

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. This is simply a reminder of what we are going to look for before merging code._
Please review the guidelines for contributing to this repository.

- [X] All work relating to the ticket is complete (Code complete/Acceptance Criteria met)
- [ ] Lint and unit tests pass locally (if appropriate)
- [X] Added tests that prove the fix is effective or that feature works (if appropriate)
- [ ] Documentation (if appropriate)

## SCREENSHOTS
[Add screenshots if any for changes you have made to the App/UI]

### CODE REVIEW

**Review by EdwardW2 — APPROVED** [review]
LGTM. Will test on the farechoice repo since some changes to the side navigation was made there for the new help screen. So going to see if this change fixes it or not.

File: KarhooUISDK/Screens/SideMenu/SideMenuContent/MenuContentPresenter.swift [hunk 1]
```diff
@@ -11,8 +11,8 @@ import KarhooSDK
 final class MenuContentScreenPresenter: MenuContentPresenter {
 
     private let routing: SideMenuHandler
-    private let mailConstructor: KarhooFeedbackEmailComposer
-    private weak var view: MenuContentView?
```
Comment (cornelkar) [inline KarhooUISDK/Screens/SideMenu/SideMenuContent/MenuContentPresenter.swift:15]: Why was the weak specifier removed?

Comment (mateid-kr) [reply KarhooUISDK/Screens/SideMenu/SideMenuContent/MenuContentPresenter.swift:15]: On the unit tests, the menu content view was being garbage collected thus being evaluated as null. Are there any considerations for the view having a weak reference ?

**Review by cornelkar — CHANGES_REQUESTED** [review]
Make the `MenuContentView` dependency `weak` again.

File: KarhooUISDK/Screens/SideMenu/SideMenuContent/MenuContentPresenter.swift (hunk 1 above)
Comment (mateid-kr) [reply KarhooUISDK/Screens/SideMenu/SideMenuContent/MenuContentPresenter.swift:15]: Done

**Pull request merged**